### DISCUSSION TITLE
Finding DKIM Selector

### DISCUSSION THREAD

**Discussion by simonmurrell1979** [MichaCo/DnsClient.NET#205] [Q&A] [answered]
Hello Michael / Community

I hope you are well!!!! Do you have any examples of how to find the DKIM selectors of a particular domain. How one would look them up. I tried to do a TXT lookup but it doesn't bring back the DKIM selector and information I am looking for. If I put in the physical DKIM selector name, I am able to extract the data I am looking for. Many thanks in advance!

**Comment by MichaConrad** [comment, marked as answer]
You can find a pretty good explanation how DKIM works here: https://www.cloudflare.com/learning/dns/dns-records/dns-dkim-record/

DKIM selectors are just domain names created by the email provider which have to match the email signature.

For security reasons, the DNS system does not allow querying all domain names configured in a zone.
You can do a zone transfer request to get everything, but to do so, you need administrative access to the source DNS server of the target domain.

**Comment by simonmurrell1979** [comment]
Ah okay as some online tools give you all this information and others you are required to insert the selector. I thought there may be an automated way using your library to be able to request all the selectors.

**Reply by MichaConrad** [reply]
No, DKIM selectors or validating an email has nothing to do with DNS per se.
The only thing you could do with this library is getting the TXT record. But you'd have to know where to get it from already.

**Reply by simonmurrell1979** [reply]
Yeah I tried via TXT and I couldn't extract the information out. I managed to get the SPF information out which is useful though. Thank you for your help. I appreciate it. I thought I was missing something. Thank you for clarifying.

**Comment by Divide-By-0** [comment]
There are websites that index/archive many of these old selectors, like [archive.prove.email](https://archive.prove.email). You can just search your domain (or hit the [API](https://archive.prove.email/api-explorer)), and get the DKIM selectors + keys that have been verified by the site. There are over a million keys on there, so not all websites, but at least most common websites should be there! You can also brute-force iterate through [selector lists](https://github.com/zkemail/registry.prove.email/files/selector_frequencies.txt) if you prefer that :)
